e-commerce/client/src/components/Checkout.tsx

```diff
@@ -0,0 +1,108 @@
+import { Redirect } from "react-router-dom";
+
+import {
+  FunctionComponent,
+  FormEventHandler,
+  useEffect,
+  useState,
+} from "react";
+import { CardElement, useStripe, useElements } from "@stripe/react-stripe-js";
+import { CartItem } from "../App";
+import { useCreatePaymentIntentMutation } from "../generated";
+
+interface CheckoutProps {
+  cartItems: CartItem[];
+  clearCart: () => void;
+}
+
+export const Checkout: FunctionComponent<CheckoutProps> = ({
+  cartItems,
+  clearCart,
+}) => {
+  const [clientSecret, setClientSecret] = useState("");
+  const [processing, setProcessing] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const [error, setError] = useState<null | string>();
+  const [succeeded, setSucceeded] = useState(false);
+  const stripe = useStripe();
+  const elements = useElements();
+  const { mutateAsync } = useCreatePaymentIntentMutation({
+    endpoint: "http://127.0.0.1:8810/graphql",
+    fetchParams: {
+      headers: {
+        "Content-type": "application/json",
+      },
+    },
+  });
+
+  const createPaymentIntent = async () => {
+    if (cartItems.length === 0) {
+      return;
+    }
+
+    const response = await mutateAsync({
+      productCartItems: cartItems.map((item) => ({
+        productId: item.product.id,
+        quantity: item.quantity,
+      })),
+    });
+
+    setClientSecret(response.createPaymentIntent);
+  };
+
+  const onSubmit: FormEventHandler = async (event) => {
+    event.preventDefault();
+
+    // show a loading state so user knows what is going
+
+    setProcessing(true);
+
+    const payload = await stripe?.confirmCardPayment(clientSecret, {
+      payment_method: {
+        card: elements?.getElement(CardElement)!,
+      },
+    });
+
+    // check payload status
+    if (payload?.error) {
+      setError(`Payment Failed`);
+      setProcessing(false);
+    } else {
+      // if succedded, redirect to thank you page
+      setError(null);
+      setProcessing(false);
+      setSucceeded(true);
```

Comment:
I don't think we need success and redirect. Here, we can just call `router.push` i think to redirect the user. Please let me know if this function exists. There should be a hook or something.

Reply:
Done